internal/scheduler/scheduling_algo.go

```diff
@@ -543,10 +541,12 @@ func (l *FairSchedulingAlgo) totalAllocationByPoolAndQueue(executors []*schedule
 			}
 			jobSchedulingInfo := job.JobSchedulingInfo()
 			if jobSchedulingInfo != nil {
-				allocation.AddResourceList(
-					int32(jobSchedulingInfo.Priority),
-					jobSchedulingInfo.GetTotalResourceRequest(),
-				)
+				priorityClass, ok := l.priorityClasses[jobSchedulingInfo.PriorityClassName]
```

Comment:
We shouldn't schedule this job. Either skip the job entirely or return an error.

Reply:
This function is called at the start of the scheduling round, when constructing a `fairShareSchedulingContext`:

https://github.com/armadaproject/armada/blob/f3fa267c875dd7908a7efc6fd62dd38a8a014a77/internal/scheduler/scheduling_algo.go#L289-L290

If I understand correctly, the jobs that we are looping over have already been scheduled; in particular, encountering an unknown priority class here means that the scheduler is in an inconsistent state.

This function could return an error instead, but I'm worried about the implications of making this change. If we abort the scheduling round here, wouldn't the next leader just run into the same issue?

Reply:
Actually, returning an error would be consistent with some of the other error handling in `scheduleOnExecutor`:

https://github.com/armadaproject/armada/blob/f3fa267c875dd7908a7efc6fd62dd38a8a014a77/internal/scheduler/scheduling_algo.go#L373-L377

On the other hand, `constructNodeDb` seems to prefer to just log and carry on:

https://github.com/armadaproject/armada/blob/f3fa267c875dd7908a7efc6fd62dd38a8a014a77/internal/scheduler/scheduling_algo.go#L435-L442